@xen-orchestra/proxy/src/app/_debounceResource.js

```diff
@@ -12,24 +14,23 @@ export async function debounceResource(pDisposable, hooks, delay = 0) {
     if (timeoutId !== undefined) {
       clearTimeout(timeoutId)
       timeoutId = undefined
-      hooks.removeListener('stop', onStop)
 
       return disposable.dispose()
     }
   }
 
-  const onStop = () => {
+  disposers.push(() => {
```

Comment:
Disposers will accumulate indefinitely even if the resource has already been disposed of.